homework/index.js

```diff
@@ -29,19 +29,124 @@
     });
     return elem;
   }
+  // table of repository's basic information
+  function createDetailsTable(index, data, container) {
```

Comment:
This function only works on a single repository. So rather than passing it through the argument list the complete array of repositories (`data`) and the `index` value which indicates which repository object to use it would be better to just call this function with the actual repository object to use, i.e.:

```js
function createDetailsTable(repo,  container) {
  ...
  createAndAppend('a', td, {
    href: repo.html_url,
    text: repo.name,
    target: '_blank',
  });
  ...
}

...

createDetailsTable(data[0], container);
...
createDetailsTable(data[index], container);
```

Also, use the name `data` only if the variable can represent any data (like internally in the `fetchJSON` function). If you have detailed knowledge about what the variable contains, use that knowledge to pick a more descriptive name. In this case, `data` contains an array of repository objects. So renaming `data` to `repositories` would take some of guessing away of what `data` represents here. (Notwithstanding as I said earlier that it is better to pass a single repository object `repo` here).

Reply:
But I still need the `index` and `data` to call the `createTd` function

Reply:
It was containing all the repos, but after putting the `createTd` function apart, it deals with only one repo.
But still need it

Reply:
I don't think so. You can call your `createTd` function like this, can't you?

```js
createTd('Description', repo.description, tbody);
```

etc.

Reply:
I did it, Thanks

Reply:
I have changed the `listContributors` also ):